src/lib.rs

```diff
@@ -46,13 +46,12 @@ pub trait DataProvider<Data> {
 
 /// The source of finalization of the units that consensus produces.
 ///
-/// The [`FinalizationProvider::data_finalized`] method is called whenever a new unit
-/// (or more precisely the data they carry) is available.
-pub trait FinalizationProvider<Data> {
-    type Error: Debug + 'static;
-
-    /// Data has been finalized. The calls to this function are ordered by the same order as the units.
-    fn data_finalized(&self, data: Data) -> Result<(), Self::Error>;
+/// The [`FinalizationHandler::data_finalized`] method is called whenever a piece of data input to the algorithm
+/// using [DataProvider::get_data] has been finalized, in order of finalization.
```

Comment:
I think you missed some backticks in the second brackets.

Reply:
Thanks